crates/xline/src/storage/auth_store/store.rs

```diff
@@ -1140,6 +1163,14 @@ where
     }
 }
 
+/// Get common name from tonic request
+fn get_cn<T>(request: &tonic::Request<T>) -> Option<String> {
+    let chain = request.peer_certs()?;
+    let cert_der = chain.first()?;
```

Comment:
Is it possiable that a certificate chain has multiple certs here? Which one represents the proper CN?

Reply:
The first one in the certificate chain is what we need